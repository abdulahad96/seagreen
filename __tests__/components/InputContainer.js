import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import InputContainer from '../../src/components/Input';


describe('Component Input Container', () => {

    const onChange = jest.fn();
    test('renders correctly', () => {
        const output = render(<InputContainer
            inputContainerStyle={{}}
            inputFieldStyle={{}}
            autoCapitalize
            isActive
            placeholderTextColor=""
            placeHolder={''}
            onChange={onChange}
        />);
        expect(output).toMatchSnapshot();
    });
    test('renders with auto capitalize', () => {
        const output = render(<InputContainer
            inputContainerStyle={{}}
            inputFieldStyle={{}}
            autoCapitalize
            isActive
            placeholderTextColor=""
            placeHolder={''}
            onChange={onChange}
        />);
        expect(output).toMatchSnapshot();
    });
    test('renders with Secure text', () => {
        const output = render(<InputContainer
            inputContainerStyle={{}}
            inputFieldStyle={{}}
            autoCapitalize
            isActive
            placeholderTextColor=""
            placeHolder={''}
            secureTextEntry
            onChange={onChange}
        />);
        expect(output).toMatchSnapshot();
    });
  });
