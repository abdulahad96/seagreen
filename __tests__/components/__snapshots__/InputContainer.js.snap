// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Input Container renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#c2c2c2",
        "borderRadius": 10,
        "borderWidth": 2,
        "marginVertical": 20,
        "width": "95%",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "width": 675,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize={true}
      onChangeText={[Function]}
      placeholder=""
      placeholderTextColor=""
      rejectResponderTermination={true}
      secureTextEntry={false}
      style={
        Array [
          Object {
            "color": "#00002a",
            "fontFamily": undefined,
            "fontSize": 32,
            "width": 562.5,
          },
          Object {
            "borderColor": "#24a790",
          },
          Object {},
        ]
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/icons/check.png",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Component Input Container renders with Secure text 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#c2c2c2",
        "borderRadius": 10,
        "borderWidth": 2,
        "marginVertical": 20,
        "width": "95%",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "width": 675,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize={true}
      onChangeText={[Function]}
      placeholder=""
      placeholderTextColor=""
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "color": "#00002a",
            "fontFamily": undefined,
            "fontSize": 32,
            "width": 562.5,
          },
          Object {
            "borderColor": "#24a790",
          },
          Object {},
        ]
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/icons/check.png",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Component Input Container renders with auto capitalize 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#c2c2c2",
        "borderRadius": 10,
        "borderWidth": 2,
        "marginVertical": 20,
        "width": "95%",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "width": 675,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize={true}
      onChangeText={[Function]}
      placeholder=""
      placeholderTextColor=""
      rejectResponderTermination={true}
      secureTextEntry={false}
      style={
        Array [
          Object {
            "color": "#00002a",
            "fontFamily": undefined,
            "fontSize": 32,
            "width": 562.5,
          },
          Object {
            "borderColor": "#24a790",
          },
          Object {},
        ]
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/icons/check.png",
          }
        }
      />
    </View>
  </View>
</View>
`;
